run_experiments.py

```diff
@@ -7,7 +7,7 @@
 # call(["python", "run.py", "-dataset-name", "cifar10", "--b","128", "--epoch", "50", "--number_of_rsacm", "1", "--distill_type", "per_rsacm"])
 # call(["python", "run.py", "-dataset-name", "cifar10", "--b","128", "--epoch", "50", "--number_of_rsacm", "8", "--distill_type", "per_accm"])
 # call(["python", "run.py", "-dataset-name", "cifar10", "--b", "256", "--epoch", "35", "--number_of_rsacm", "8", "--distill_type", "per_rsacm", '--arch', 'resnet50'])
-call(["python", "run.py", "-dataset-name", "cifar10", "--b", "256", "--epoch", "35", "--number_of_rsacm", "8", "--distill_type", "per_rsacm", '--arch', 'resnet50', '--inference', '--load_student'])
+call(["python", "run.py", "-dataset-name", "cifar10", "--b", "16", "--epoch", "35", "--number_of_rsacm", "8", "--distill_type", "per_rsacm", '--arch', 'resnet50', '--load_student'])
```

Comment:
After adding gating networks I had to drastically lower the batch_size - on batch_size 256 getting out-of-memory errors.

Reply:
I see no reason why adding gating networks should increase the amount of needed memory by that much. That's suspicious, gating networks are very small, so perhaps we somehow have a reference to tensors from previous iterations?

Reply:
I was able to set batch_size back to `256`, after setting the 'cudnn.benchmark' to False. https://discuss.pytorch.org/t/what-does-torch-backends-cudnn-benchmark-do/5936

Reply:
Interesting. If I understood this thread correctly, the people there say that cuda kernels are cached if inputs sizes vary. Shouldn't we have a constant image size in each module? This thread: https://discuss.pytorch.org/t/set-torch-backends-cudnn-benchmark-true-consumes-huge-amount-of-memory/131010 suggests that cudnn.benchmark=True causes pytorch to choose an algorithm with huge memory requirements, which is more convincing explanation. We should remember this, it may be important later on.